src/modm/platform/i2c/sam_x7x/i2c_master.hpp.in

```diff
@@ -0,0 +1,180 @@
+		if constexpr (baudrate > minLowTimeLimit) {
+			// calculate ideal low and high prescaler values (formula from ASF vendor HAL)
+			constexpr auto cldiv = uint32_t(std::round((minLowTime * clock) - 3));
+			constexpr auto tHigh = 1.f / ((baudrate + (baudrate - minLowTimeLimit)) * 2.f);
+			constexpr auto chdiv = uint32_t(std::round((tHigh * clock) - 3));
+
+			// use 2^N pre-divider if max. prescaler exceeds 8 bits
+			constexpr auto ckdiv = std::max(0, std::max(bits(cldiv), bits(chdiv)) - 8);
+			constexpr uint32_t cldivScaled = std::round(float(cldiv) / (1 << ckdiv));
+			constexpr uint32_t chdivScaled = std::round(float(chdiv) / (1 << ckdiv));
+
+			if (!checkBaudrate<clock, baudrate, tolerance, ckdiv, cldivScaled, chdivScaled>())
+				return std::nullopt;
+
+			return TWIHS_CWGR_HOLD(hold) | TWIHS_CWGR_CLDIV(cldivScaled) |
+				TWIHS_CWGR_CHDIV(chdivScaled) | TWIHS_CWGR_CKDIV(ckdiv);
+		} else {
+			constexpr auto div = uint32_t(std::round(clock / (baudrate * 2.f) - 3));
+
+			// use 2^N pre-divider if max. prescaler exceeds 8 bits
+			constexpr auto ckdiv = std::max(0, bits(div) - 8);
+			constexpr uint32_t divScaled = std::round(float(div) / (1 << ckdiv));
+
+			if (!checkBaudrate<clock, baudrate, tolerance, ckdiv, divScaled, divScaled>())
+				return std::nullopt;
+
+			return TWIHS_CWGR_HOLD(hold) | TWIHS_CWGR_CLDIV(divScaled) |
+				TWIHS_CWGR_CHDIV(divScaled) | TWIHS_CWGR_CKDIV(ckdiv);
+		}
+	}
+
+public:
+	template<class... Signals>
+	static void
+	connect(PullUps pullups = PullUps::External, ResetDevices reset = ResetDevices::Standard)
+	{
+		using Scl = GetPin_t<PeripheralPin::Twck, Signals...>;
+		using Sda = GetPin_t<PeripheralPin::Twd, Signals...>;
+
+		using Peripheral = Peripherals::Twihs<{{ id | int }}>;
+
+		using SclConnector = typename Scl::template Connector<Peripheral, Peripheral::Twck<{{ id | int }}>>;
+		using SdaConnector = typename Sda::template Connector<Peripheral, Peripheral::Twd<{{ id | int }}>>;
+
+		const InputType input =
+			(pullups == PullUps::Internal) ? InputType::PullUp : InputType::Floating;
+
+		Scl::configure(input);
+		Sda::configure(input);
+		if (reset != ResetDevices::NoReset) resetDevices<Scl>(uint32_t(reset));
+		SclConnector::connect();
+		SdaConnector::connect();
+	}
+
+	/**
+	 * Set up the I2C module for master operation.
+	 *
+	 * @param	rate
+	 *		`Standard` (100 kHz) or `Fast` (400 kHz)
+	 */
+	template<class SystemClock, baudrate_t baudrate=kBd(100), percent_t tolerance=pct(5)>
+	static void
+	initialize()
+	{
+		static_assert(baudrate <= 400'000, "Baudrate must not exceed 400 kHz for I2C fast mode");
+		constexpr std::optional<uint32_t> registerValue = calculateTimings<SystemClock, baudrate, tolerance>();
+		static_assert(bool(registerValue), "Could not find a valid clock configuration for the requested"
+			" baudrate and tolerance");
+
+		initializeWithClockConfig(registerValue.value());
+	}
+
+	// start documentation inherited
```

Comment:
I am not really sure it does anything. It's there in the other I²C drivers as well. If it doesn't work all drivers need to be fixed. I'd consider that out of scope for this PR.

Reply:
This "command" does nothing, it's not even parsed by doxygen/doxypress because the comment does not start with `///`.

There is a [switch in doxypress](https://www.copperspice.com/docs/doxypress/project-file.html#inherit-docs) (and also in doxygen) to inherit the documentation from documented members which is re-implemented and our config has it enabled https://github.com/modm-io/modm/blob/04ed0a52e03964811eb9764306bb1b61de5b5f60/tools/doc_generator/doxypress.json.in#L48
but it does not seem to work 😒

---

In my opinion we should remove all occurrences of `start documentation inherited`, but in the scope of another PR.